packages/circuit-ui/components/Table/components/TableHeader/TableHeader.tsx

```diff
@@ -225,45 +176,41 @@ const StyledHeader = styled('th', {
 );
 
 /**
- * TableHeader component for the Table. You shouldn't import this component
- * directly, the Table handles it
+ * TableHeader for the Table component. The Table handles rendering it.
  */
 const TableHeader: FC<TableHeaderProps> = ({
-  sortLabel,
-  sortable = false,
   children,
-  sortDirection,
   condensed,
-  onClick,
   align = 'left',
   scope = 'col',
   fixed = false,
   isHovered = false,
-  isSorted = false,
+  sortParams = { sortable: false },
   ...props
-}) => {
-  const label = getLabelSort(sortable, sortDirection, sortLabel);
-  return (
-    <StyledHeader
-      sortable={sortable}
-      condensed={condensed}
-      aria-sort={getAriaSort(sortable, sortDirection)}
-      aria-label={label}
-      onClick={onClick}
-      align={align}
-      scope={scope}
-      fixed={fixed}
-      isHovered={isHovered}
-      isSorted={isSorted}
-      data-testid="table-header"
-      {...props}
-    >
-      {sortable && (
-        <SortArrow label={label} direction={sortDirection} onClick={onClick} />
-      )}
-      {children}
-    </StyledHeader>
-  );
-};
+}) => (
+  <StyledHeader
+    condensed={condensed}
+    align={align}
+    scope={scope}
+    fixed={fixed}
+    isHovered={isHovered}
+    sortable={sortParams.sortable}
+    isSorted={!!sortParams.isSorted}
+    aria-label={sortParams.sortLabel}
+    aria-sort={
+      sortParams.sortable ? sortParams.sortDirection || 'none' : undefined
+    }
+    data-testid="table-header"
+    {...props}
+  >
+    {sortParams.sortable && (
+      <SortArrow
+        label={sortParams.sortLabel}
+        direction={sortParams.sortDirection}
+      />
```

Comment:
Where did the `onClick` go? 🤔

Reply:
I don't think we actually need it, because it's already set on the entire parent `<th>`. Is it better to also keep it on this button?

Reply:
Ah, I didn't see that. I think this is still important for accessibility. I assume the SortArrow is rendered as a `<button>` and is focusable, so when operated with a keyboard, it needs the `onClick` to trigger the action, right?

Reply:
Well the entire table header is focusable because the entire area is clickable. I am tempted to make the button inside `aria-hidden` and give a button role to the entire <th> if it doesn't have one yet, What do you think?

![Screenshot from 2021-06-07 21-29-14](https://user-images.githubusercontent.com/35560568/121077533-0410b800-c7d8-11eb-82b5-730a3d0b1da8.png)

Reply:
clickable !== focusable

When you tab to a sortable table head and log the focused element, you'll see that it's actually the button inside that is has focus:

<img width="1440" alt="Screenshot 2021-06-07 at 23 31 15" src="https://user-images.githubusercontent.com/11017722/121090173-75f0fd80-c7e8-11eb-8452-a47968743215.png">

The entire cell looks focused because of the use of `:focus-within` [here](https://github.com/sumup-oss/circuit-ui/blob/main/packages/circuit-ui/components/Table/components/TableHeader/TableHeader.tsx#L159-L181).

I checked the Vercel preview and I'm still able to trigger the sorting with my keyboard 🤔 I'd still keep the `onClick` on the button in addition to the table head though.

Reply:
Also, that's a weird bug in the screenshot above: the floating column isn't visible 👀

Reply:
Added the onClick back in 724d31f.

About the weird bug, I've found that it was related to the columns' `overflow-x:auto;` on responsive styles with a fixed row headers column, but haven't found a way to address it yet.

Since it seems to only be affecting Storybook, I would suggest keeping it out of scope to move forward with this PR, and address it at some other time (we also have other issues with `Table` stories, like scrollable styles)